text/0000-local-snapshot-file-format/0000-local-snapshot-file-format.md

```diff
@@ -0,0 +1,186 @@
+* delete transaction data below the given milestone
+
+Current node implementations use a [local snapshot file format](https://github.com/iotaledger/iri-ls-sa-merger/tree/351020d3b5e342b6e9a41f2868575ab7ff8c251c#generating-an-export-file-from-a-localsnapshots-db) which only works with account based ledgers. For Chrysalis Phase 2 this file format has to be assimilated to support a UTXO based ledger.
+
+# Detailed design
+
+All types are serialized in little-endian and occur in the sequence of the rows defined below. Local snapshot files are compressed via zlib to further reduce size.
+
+SEP = solid entry point. `Array[T]` are prefixed with a varint denoting the length.
+
+This format describes version 1:
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>Version</td>
+        <td>byte</td>
+        <td>
+        The version of the local snapshot file format.
+        </td>
+    </tr>    
+    <tr>
+        <td>Milestone Index</td>
+        <td>uint64</td>
+        <td>
+        The index of the milestone of the local snapshot.
+        </td>
+    </tr>
+    <tr>
+        <td>Milestone Hash</td>
+        <td>ByteArray[32]</td>
+        <td>
+        The BLAKE2b-256 hash of the milestone payload.
+        </td>
+    </tr>
+    <tr>
+        <td>Timestamp</td>
+        <td>uint64</td>
+        <td>
+        The UNIX epoch timestamp in seconds of when this snapshot was created.
+        </td>
+    </tr>
+    <tr>
+        <td>SEPs</td>
+        <td>Array[ByteArray[32]]</td>
+        <td>
+        The BLAKE2b-256 hashes of the SEP messages at the cut off point of the given milestone.
+        </td>
+    </tr>
+    <tr>
+        <td>UTXOs</td>
+        <td colspan="2">
+            <details open="true">
+                <summary>Array[UTXO]</summary>
+                <blockquote>
+                Describes the unspent transaction outputs per transaction.
+                </blockquote>
+                <table>
+                    <tr>
+                        <td><b>Name<b></td>
+                        <td><b>Type</b></td>
+                        <td><b>Description</b></td>
+                    </tr>
+                    <tr>
+                        <td>Transaction Hash</td>
+                        <td>ByteArray[32]</td>
+                        <td>The BLAKE2b-256 hash of the transaction.</td>
+                    </tr>
+                    <tr>
+                        <td>
+                        Unspent Outputs
+                        </td>
+                        <td colspan="2">
+                            <details open="true">
+                                <summary>Array[Outputs]</summary>
+                                <table>
+                                    <tr>
+                                        <td>Index</td>
+                                        <td>byte</td>
```

Comment:
This should be a `varint` to be consistent with [Inputs / Outputs Count fields](https://github.com/luca-moser/protocol-rfcs/blob/signed-tx-payload/text/0000-signed-transaction-payload/0000-signed-transaction-payload.md#serialized-layout), even if we "informally" define such count to be < 127.

Reply:
I guess this needs to be revisited once, when exactly that part of RFC #18 has been finalized. `byte` definitely seems to restricting to me.

Reply:
All array lengths will be replaced with fixed sized integers.